### PR TITLE
rest: add support for protobuf to remaining endpoints that support a binary content type

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Comment by bmwill** [comment]
After talking with @wlmyng I'll add a commit on top of this to have the client request BCS data till this rolls out to all networks

File: crates/sui-rest-api/src/client/mod.rs [hunk 1]
```diff
@@ -53,12 +53,13 @@ impl Client {
...
         let request = self.inner.client().get(url);
 
-        let proto = self
-            .inner
-            .protobuf::<crate::proto::FullCheckpoint>(request)
-            .await?
-            .into_inner();
-        proto.try_into().map_err(Into::into)
+        self.inner.bcs(request).await.map(Response::into_inner)
+        // let proto = self
```
Comment (wlmyng) [inline crates/sui-rest-api/src/client/mod.rs:57]: Nit: comment

File: crates/sui-rest-api/src/proto/mod.rs [hunk 2]
```diff
@@ -337,6 +337,36 @@ impl TryFrom<&UserSignature> for sui_types::signature::GenericSignature {
     }
 }
 
+//
```
Comment (wlmyng) [inline crates/sui-rest-api/src/proto/mod.rs:340]: Is this file translations between various types

**Review by wlmyng — APPROVED** [review]

**Review by sadhansood — APPROVED** [review]
I think the 2 step rollout makes sense i.e. server followed by client

**Comment by bmwill** [comment]
> I think the 2 step rollout makes sense i.e. server followed by client

Yeah, that and while I think these protos are a good start I think it may be good to sit on them for a short period of time (a week or two) to make sure there aren't any other changes we want to make before we can't trivially tweak them.

**Pull request merged**